### PR TITLE
Robust ensemble when no memory

### PR DESCRIPTION
**Pull request by franchuterivera**
Make sure that if a memory error occurs, because the user didn't specify memory correctly, at least an ensemble of the best model is returned.

ensemble_size takes priority and can be used to change the behavior

### CODE REVIEW

File: test/test_ensemble_builder/test_ensemble.py [hunk 1]
```diff
@@ -463,3 +466,62 @@ def testPredict(self):
...
+                    'num_run': 6,
+                    'configuration_origin': 'Random Search'}
+            ),
+            status=None,
+            origin=None,
+        )
+        ensemble = SingleModelEnsemble(
+             ensemble_size=5,
+             task_type=None,
+             metric=accuracy,
```
Comment (mfeurer) [inline test/test_ensemble_builder/test_ensemble.py:515]: Could you please add a test with a loss function (i.e. logloss), too?

Comment (franchuterivera) [reply test/test_ensemble_builder/test_ensemble.py:515]: Added a check for this. BTW, thanks for flagging this, I was able to find a bug in the code due to this, which has been corrected. Also added extra checks to capture this.

File: autosklearn/ensembles/singlemodel_ensemble.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            if score > best_model_score:
+                best_model_identifier = [
+                    (self.random_state, run_value.additional_info['num_run'], run_key.budget)
+                ]
+                best_model_score = score
+
+        if best_model_identifier is None:
+            raise ValueError(
+                "No valid model found in run history. This means smac was not able to fit"
+                " a valid model. Please check the run file for errors."
```
[the remark this answers is not in the document]
Comment (franchuterivera) [reply autosklearn/ensembles/singlemodel_ensemble.py]: Added in new commit

**Comment by franchuterivera** [comment]
I tested with Python 3.7.7 fit_pSMAC due to travis failures but it looks ok (no issue on my side or in other version of python)

File: test/test_automl/test_automl.py [hunk 3]
```diff
@@ -503,40 +503,48 @@ class MyException(Exception):
...
+                mock_ensemble_run.side_effect = MemoryError
+                automl.fit(
+                    X_train, Y_train, task=MULTICLASS_CLASSIFICATION,
+                )
 
-        # The load model is robust to this and loads the best model
-        automl._load_models()
-        self.assertTrue(automl.ensemble_ is not None)
+            # A memory error occurs in the ensemble construction
+            self.assertTrue(automl._backend.load_ensemble(automl._seed) is None)
```
Comment (mfeurer) [inline test/test_automl/test_automl.py:528]: Please use `AssertIsNone` here

File: test/test_automl/test_automl.py [hunk 4]
```diff
@@ -503,40 +503,48 @@ class MyException(Exception):
...
 
-        # Just 1 model is there for ensemble and all weight must be on it
-        get_models_with_weights = automl.get_models_with_weights()
-        self.assertEqual(len(get_models_with_weights), 1)
-        self.assertEqual(get_models_with_weights[0][0], 1.0)
+            # The load model is robust to this and loads the best model
+            automl._load_models()
+            self.assertTrue(automl.ensemble_ is not None)
```
Comment (mfeurer) [inline test/test_automl/test_automl.py:532]: Same here

**Pull request merged**